llama.h

```diff
@@ -71,28 +71,26 @@ extern "C" {
 
     typedef void (*llama_progress_callback)(float progress, void *ctx);
 
-    struct llama_context_params {
-        int n_ctx;                             // text context
-        int n_batch;                           // prompt processing batch size
-        int n_gpu_layers;                      // number of layers to store in VRAM
-        int main_gpu;                          // the GPU that is used for scratch and small tensors
-        float tensor_split[LLAMA_MAX_DEVICES]; // how to split layers across multiple GPUs
+   struct llama_context_params {
         bool low_vram;                         // if true, reduce VRAM usage at the cost of performance
-        int seed;                              // RNG seed, -1 for random
-
         bool f16_kv;     // use fp16 for KV cache
         bool logits_all; // the llama_eval() call computes all logits, not just the last one
         bool vocab_only; // only load the vocabulary, no weights
         bool use_mmap;   // use mmap if possible
         bool use_mlock;  // force system to keep model in RAM
         bool embedding;  // embedding mode only
 
```

Comment:
I believe we need to pad the `bool`s up to 8 bytes with:

```c
char padding[1];
```

Alternatively, move the `bool`s at the end of the struct.

Would appreciate comment from someone more knowledgable what is best practice here, as I am not 100% sure

Reply:
I went with moving them at the bottom, let me know if you prefer the other way around. I'm not entitled to give a better advice here, but maybe @lu-zero can likely give some input here too?

Reply:
I'd also open an issue on gcc/nvidia/google since it is very unexpected and I cannot find a reason why it is happening.

Given how rare is to pass structs by value in C it should be a bug not caught. I'd advise to pass by pointer to avoid suffering further in the future.

Reply:
suggest to use pack attribute as this is not performance critial while it is part of interface.